packages/fxa-auth-server/lib/routes/totp.js

```diff
@@ -155,6 +155,7 @@ module.exports = (log, db, mailer, customs, config, glean) => {
         // See #5154.
         await db.verifyTokensWithMethod(sessionToken.id, 'email-2fa');
 
+        await profileClient.deleteCache(uid);
         await log.notifyAttachedServices(
```

Comment:
Is there a scenario where we would call `notifyAttachedServices` with `profileDataChange` and not want to clear the cache? Does it make sense to put it inside of that function?

Reply:
Nope! And in fact this is what I initially set out to do, but I encountered a couple small things that made me go with this approach instead (at least for now).

First, the profileClient takes the logger as a dependency, so having the logger take the profileClient as a dependency would have created a two way dependency, which I wanted to avoid. 

Second, I don’t think NotifyService should be in the logger at all. I can see why it’s there by looking at how the code evolved, but it doesn’t feel like a good separation of concerns to me.

Going forwards, I think we could do what you’re suggesting, but we’d want to add ProfileClient in our libs. Then we could make the NotifierSerivce depend on profile client. It’d be nice to have the profile client in libs anyways, since there’s already some copypasta in graphql-api, admin-server, and auth-server going. I filed [FXA-10368](https://mozilla-hub.atlassian.net/browse/FXA-10368), so we don’t lose track of this feedback.